packages/experimental/Link/src/Link.tsx

```diff
@@ -0,0 +1,53 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { View } from 'react-native';
+import { linkName, LinkType, LinkProps, ILinkState } from './Link.types';
+import { compose, mergeProps, withSlots, UseSlots } from '@fluentui-react-native/framework';
+import { useLink } from './useLink';
+import { TextV1 as Text } from '@fluentui-react-native/text';
+import { stylingSettings } from './Link.styling';
+
+/**
+ * A function which determines if a set of styles should be applied to the component given the current state and props of the link.
+ *
+ * @param layer The name of the state that is being checked for
+ * @param state The current state of the link
+ * @param userProps The props that were passed into the link
+ * @returns Whether the styles that are assigned to the layer should be applied to the link
+ */
+export const linkLookup = (layer: string, state: ILinkState, userProps: LinkProps): boolean => {
+  return (
+    state[layer] ||
+    userProps[layer] ||
+    layer === userProps['appearance'] ||
+    (layer === 'inline' && userProps.inline) ||
+    (layer === 'disabled' && userProps.disabled)
+  );
+};
+
+export const Link = compose<LinkType>({
+  displayName: linkName,
+  ...stylingSettings,
+  slots: {
+    root: View,
+    content: Text,
+  },
+  useRender: (userProps: LinkProps, useSlots: UseSlots<LinkType>) => {
+    const link = useLink(userProps);
+    // grab the styled slots
+    const Slots = useSlots(userProps, (layer) => linkLookup(layer, link.state, userProps));
+    // now return the handler for finishing render
+    return (final: LinkProps, ...children: React.ReactNode[]) => {
+      // pull onPress out to let Link's Pressable deal with all press events
```

Comment:
I don't fully understand what this is doing, do you mind explaining in more detail how this works when this Link doesn't have a Pressable?

Reply:
I had to prevent passing in onPress onto Text slot because it was getting called twice as an inline link.

Reply:
This may also change with my upcoming pr with pressable text.

Reply:
Oh okay so pressable here is native, not the RN Pressable

Reply:
Also does that mean that onPress should be passed along in the non-inline case?

Reply:
Even for non-inline cases, onPress gets handled on the JS side so everything seems to work fine with the current implementation.

Reply:
I rephrased the comment.